src/pal/src/exception/seh.cpp

```diff
@@ -396,6 +396,8 @@ bool CatchHardwareExceptionHolder::IsEnabled()
 
 #ifdef __llvm__
 __thread 
+#elif defined(__GNUC__)
```

Comment:
clang defines `__GNUC__` too, so you can just change the `__llvm__` to `__GNUC__` without an extra `#ifdef`